test/functional/feature_index_prune.py

```diff
@@ -19,9 +20,25 @@ def set_test_params(self):
             ["-fastprune", "-prune=1", "-blockfilterindex=1"],
             ["-fastprune", "-prune=1", "-coinstatsindex=1"],
             ["-fastprune", "-prune=1", "-blockfilterindex=1", "-coinstatsindex=1"],
-            []
+            [],
         ]
 
+    def setup_network(self):
+        self.setup_nodes()  # No P2P connection, so that linear_sync works
+
+    def linear_sync(self, node_from, *, height_from=None):
+        # Linear sync over RPC, because P2P sync may not be linear
```

Comment:
I think a little more detail would be helpful since it may not be obvious what the issue is.

``` suggestion
# Linear sync over RPC, because P2P sync may lead to blocks being received and saved to disk out of order which means prune heights would non-deterministic.
```